netket/vqs/mc/mc_state/state.py

```diff
@@ -356,6 +372,57 @@ def n_discard(self, val) -> int:
         )
         self.n_discard_per_chain = val
 
+    @property
+    def chunk_size(self) -> int:
+        """
+        Suggested *maximum size* of the chunks used in forward and backward evaluations
+        of the Neural Network model. If your inputs are smaller than the chunk size
+        this setting is ignored.
+
+        This can be used to lower the memory required to run a computation with a very
+        high number of samples or on a very large lattice. Notice that inputs and
+        outputs must still fit in memory, but the intermediate computations will now
+        require less memory.
+
+        This option comes at an increased computational cost. While this cost should
+        be negligible for large-enough chunk sizes, don't use it unless you are memory
+        bound!
+
+        This option is an hint: only some operations support chunking. If you perform
+        an operation that is not implemented with chunking support, it will fall back
+        to no chunking. To check if this happened, set the environment variable
+        `NETKET_DEBUG=1`.
+        """
+        return self._chunk_size
+
+    @chunk_size.setter
+    def chunk_size(self, chunk_size: int):
+        # disable chunks if it is None
+        if chunk_size is None:
+            self._chunk_size = None
+            return
+
+        if chunk_size <= 0:
+            raise ValueError("Chunk size must be a positive integer. ")
+
+        if not _is_power_of_two(chunk_size):
+            warnings.warn(
+                "For performance reasons, we suggest to use a power-of-two chunk size."
+            )
+
+        if chunk_size >= self.n_samples_per_rank:
+            # no problem!
+            pass
+        elif (self.n_samples_per_rank % chunk_size) != 0:
+            raise ValueError(
+                f"chunk_size={chunk_size}`<`n_samples_per_rank={self.n_samples_per_rank}, "
+                "chunk_size is not an integer fraction of `n_samples_per rank`. This is"
+                "unsupported. Please change `chunk_size` so that it divides evenly the"
+                "number of samples per rank."
+            )
```

Comment:
This may be replaced with a call to `compute_chain_length` which checks this (it only really saves duplicating the error message though)
```suggestion
        compute_chain_length(self.sampler.n_chains, self.n_samples, chunk_size)
```

Reply:
I implemented this change, but i disentangled the check for chunk size into a different fucntion